src/main/java/fr/moribus/imageonmap/commands/maptool/UpdateCommand.java

```diff
@@ -83,50 +153,85 @@ protected void run() throws CommandException {
             default:
                 scaling = ImageUtils.ScalingType.CONTAINED;
         }
-        ImageMap map;
-        if (scaling.equals(ImageUtils.ScalingType.CONTAINED)) {
-            map = getMapFromArgs(player, 1);
-        } else {
-            map = getMapFromArgs(player, 2);
-        }
-        try {
-            url = new URL(args[0]);
-            MapManager.load();
 
-            Integer[] size = {1, 1};
-            if (map.getType() == ImageMap.Type.POSTER) {
-                size = map.getSize(new HashMap<String, Object>(), map.getUserUUID(), map.getId());
+        //TODO passer en static
+        ImageOnMap.getPlugin().getCommandWorker().offlineNameFetch(playerName, uuid -> {
+            if (uuid == null) {
+                if (playerSender != null) {
+                    info(playerSender, I.t("The player {0} does not exist.", playerName));
+                } else {
+                    PluginLogger.info(I.t("The player {0} does not exist.", playerName));
+                }
+                return;
             }
-            //assert size != null;
-            int width = size[0];
-            int height = size[1];
+            ImageMap map = MapManager.getMap(uuid, mapName);
+
+            if (map == null) {
+                if (playerSender != null) {
+                    info(playerSender, I.t("This map does not exist."));
+                } else {
+                    PluginLogger.info(I.t("This map does not exist."));
+                    ;
+                }
```

Comment:
Idem pas de séparation entre les cas. Et idem sur `finalSender`.
```suggestion
                warning(sender, I.t("This map does not exist."));
```